.github/workflows/build_and_test.yml

```diff
@@ -111,8 +111,18 @@ jobs:
       - name: Get postgres headers
         run: make postgres-headers -j$(nproc)
 
-      - name: Run cargo clippy
-        run: ./run_clippy.sh
+      - uses: taiki-e/install-action@cargo-hack
+
+      # cargo hack runs the given cargo subcommand (clippy in this case) for all feature combinations.
+      # This will catch compiler & clippy warnings in all feature combinations.
+      # TODO: use cargo hack for build and test as well, but, that's quite expensive.
+      # NB: keep clippy args in sync with ./run_clippy.sh
+      - run: |
+          echo "CLIPPY_COMMON_ARGS=--locked --workspace --all-targets -- -A unknown_lints -D warning" >> $GITHUB_ENV
+      - name: Run cargo clippy (debug)
+        run: cargo hack --feature-powerset clippy $CLIPPY_COMMON_ARGS
+      - name: Run cargo clippy (release)
+        run: cargo hack --feature-powerset clippy $CLIPPY_COMMON_ARGS --release
```

Comment:
Had not thought about release, luckily we don't have `cfg(not(debug_assertions))` or `cfg(release)`.

Probably a good idea.

Reply:
Resolving needless.

Reply:
Unresovling because apparently it does not exist (didn't check that).

Reply:
```
it does not exist 
```

what is `it` here?

Reply:
`--release`

Reply:
Ok, I just made a mistake of failing to run `cargo clippy --release` locally (somehow) and got confused, thought it wasn't an option, which would had been strange.

Now `cargo clippy --release` works, so I must've fumbled something else. We should be clippy'ing the `--release` as it is supported, it was a good idea, resolving.